tests/test_data_cleaner.py

```diff
@@ -96,3 +99,114 @@ def current_size():
             self.assertTrue(removed == [])
 
             shutil.rmtree(temp_folder)
+
+    def test_early_termination(self):
+        temp_folder = tempfile.mkdtemp(suffix='_test_4')
+        sub_folder_1 = tempfile.mkdtemp(dir=temp_folder)
+        sub_folder_2 = tempfile.mkdtemp(dir=temp_folder)
+        sub_folder_3 = tempfile.mkdtemp(dir=temp_folder)
+
+        current_time = time.time()
+
+        def current_size(folder):
+            try:
+                folder_size = {temp_folder: 1024*2,
```

Comment:
I think I get the idea that if early termination didn't work then it would try to access the wrong key in the dictionary which would raise an Exception, is that true? If so let's comment thoroughly this test case for other people to understand it quicker